src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,209 @@
+        private _requestSender: RequestSender,
+        private _wepayRiskClient: WepayRiskClient,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _checkoutActionCreator: CheckoutActionCreator
+
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId, chasepay: chasepayCheckoutOptions } = options;
+
+        if (!chasepayCheckoutOptions ) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.chasepayCheckoutOptions" argument is not provided.');
+        }
+
+        const {
+            onSuccess = (payload: any) => {},
+            onCancel = () => {},
+            signInButton,
+            container,
+        } = chasepayCheckoutOptions;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+                if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                const cart = state.cart.getCart();
+                const storeConfig = state.config.getStoreConfig();
+
+                this._wepayRiskClient.initialize();
+
+                if (!cart || !storeConfig || !this._paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                return this._chasePayScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(({ ChasePay }) => {
+                        const { COMPLETE_CHECKOUT, CANCEL_CHECKOUT } = ChasePay.EventType;
+
+                        if (ChasePay.isChasePayUp && document.getElementById(container)) {
+                            ChasePay.insertBrandings({
+                                containers: [container],
+                                color: 'white',
+                            });
+                        }
+
+                        ChasePay.on(CANCEL_CHECKOUT, () => onCancel());
+
+                        ChasePay.on(COMPLETE_CHECKOUT, (payload: ChasePaySuccessPayload) => {
+                            const state = this._store.getState();
+                            const method = state.paymentMethods.getPaymentMethod(methodId);
+                            const requestId = method && method.initializationData && method.initializationData.merchantRequestId;
+
+                            this._setExternalCheckoutData(payload, requestId).then(() => {
+                                this._paymentInstrumentSelected().then(() => onSuccess(payload));
+                            });
+                        });
+
+                        ChasePay.on(ChasePay.EventType.START_CHECKOUT, () => {
```

Comment:
`START_CHECKOUT` is called when `ChasePay.startCheckout(digitalSessionId);` is executed.

Reply:
Ok, yes that's why it was confusing because you had `startCheckout` call inside `START_CHECKOUT` callback. But in order to trigger the callback, you had to call `showLoadingAnimation` / `startCheckout`. So I wasn't sure what was going on.

Reply:
Your latest commit seems to address the double `startCheckout` issue.